server/leafnode_test.go

```diff
@@ -869,3 +869,66 @@ func TestLeafCloseTLSConnection(t *testing.T) {
 	leaf.closeConnection(SlowConsumerWriteDeadline)
 	ch <- true
 }
+
+type captureDebugErrorLogger struct {
+	DummyLogger
+	errCh chan string
+}
+
+func (l *captureDebugErrorLogger) Debugf(format string, v ...interface{}) {
+	select {
+	case l.errCh <- fmt.Sprintf(format, v...):
+	default:
+	}
+}
+
+func TestLeafNodeRemoteWrongPort(t *testing.T) {
+	port := 8786
```

Comment:
Test could be simplified and use only random port.
Start a server that accepts leafnode connections (no remote defined). Then create an option with a remote leafnode that points to the client port. This can be done this way:
```
u, _ := url.Parse(fmt.Sprintf("nats://%s:%d", oa.Host, oa.Port)
ob.LeafNodes.Remotes = []*RemoteLeafOpts{{URLs:[]*url.URL{u}}}
```
above assume that oa is the options for the server accepting leafnode connections, and ob is for the server creating the leafnode connection to the client port.